src/core/AudioBusHandle.cpp

```diff
@@ -0,0 +1,254 @@
+/*
+ * AudioBusHandle.cpp - ThreadableJob between PlayHandle and MixerChannel
+ *
+ * Copyright (c) 2004-2014 Tobias Doerffel <tobydox/at/users.sourceforge.net>
+ * Copyright (c) 2025 Johannes Lorenz <jlsf2013$users.sourceforge.net, $=@>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include <QMutexLocker>
+
+#include "AudioBusHandle.h"
+#include "AudioDevice.h"
+#include "AudioEngine.h"
+#include "EffectChain.h"
+#include "Mixer.h"
+#include "Engine.h"
+#include "MixHelpers.h"
+#include "BufferManager.h"
+
+namespace lmms
+{
+
+AudioBusHandle::AudioBusHandle(const QString& name, bool hasEffectChain,
+	FloatModel* volumeModel, FloatModel* panningModel,
+	BoolModel* mutedModel) :
+	m_bufferUsage(false),
+	m_buffer(BufferManager::acquire()),
+	m_extOutputEnabled(false),
+	m_nextMixerChannel(0),
+	m_name(name),
+	m_effects(hasEffectChain ? new EffectChain(nullptr) : nullptr),
+	m_volumeModel(volumeModel),
+	m_panningModel(panningModel),
+	m_mutedModel(mutedModel)
+{
+	Engine::audioEngine()->addAudioBusHandle(this);
+	setExtOutputEnabled(true);
+}
+
+
+
+
+AudioBusHandle::~AudioBusHandle()
+{
+	setExtOutputEnabled(false);
+	Engine::audioEngine()->removeAudioBusHandle(this);
+	BufferManager::release(m_buffer);
+}
+
+
+
+
+void AudioBusHandle::setExtOutputEnabled(bool enabled)
+{
+	if(enabled != m_extOutputEnabled)
```

Comment:
I think we need a space before the parentheses here. This applies to the rest of the file as well.

```suggestion
	if (enabled != m_extOutputEnabled)
```